src/shims/unix/solarish/foreign_items.rs

```diff
@@ -26,26 +26,32 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
                 // THREAD_NAME_MAX allows a thread name of 31+1 length
                 // https://github.com/illumos/illumos-gate/blob/7671517e13b8123748eda4ef1ee165c6d9dba7fe/usr/src/uts/common/sys/thread.h#L613
                 let max_len = 32;
-                let res = this.pthread_setname_np(
+                let res = match this.pthread_setname_np(
```

Comment:
```suggestion
                // See https://illumos.org/man/3C/pthread_setname_np for the error codes.
                let res = match this.pthread_setname_np(
```

Reply:
Yet another comment that was marked as resolved without actually being resolved.

Reply:
It was automatically resolved when I applied `5dbf95bfd27066bf64711bdd5c53e592f8c6f727` with your suggestions, but then I rebased the pull request and lost it. I know that I'm quite unfocused lately and try to improve that, but that's not easy to do. Sorry!

Reply:
I recommend never using `git push --force`, and always `--force-with-lease`. That can help prevent mistakes like that.

You can set up an alias to make this easier, as `git pushf`:
```
[alias]
        pushf = push --force-with-lease
```